spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskExecutionService.java

```diff
@@ -285,6 +305,28 @@ public long executeTask(String taskName, Map<String, String> taskDeploymentPrope
 		return taskExecution.getExecutionId();
 	}
 
+	private TaskExecutionInformation findOrCreateTaskExecutionInformation(String taskName, Map<String, String> taskDeploymentProperties, String composedTaskRunnerName) {
+
+		TaskExecutionInformation taskExecutionInformation;
+		try {
+			 taskExecutionInformation = taskExecutionInfoService
+					.findTaskExecutionInformation(taskName, taskDeploymentProperties, composedTaskRunnerName);
+
+		} catch (NoSuchTaskDefinitionException e) {
+			if (autoCreateTaskDefinitions) {
+				logger.info("Creating a Task Definition {} for registered app name {}", taskName, taskName);
+				TaskDefinition taskDefinition = new TaskDefinition(taskName, taskName);
```

Comment:
Not sure if this is a valid assumption to have both the task name and the definition to be of the same.

Perhaps, we can introduce a new task deployment property (something like `spring.cloud.dataflow.task.definition`) and infer the task definition from the task deployment property.

Reply:
Hi @ilayaperumalg. There are a couple of issues with your suggestion. The main one is that `taskName` in the launch request is the task name. I don't think we should overload the meaning of that field. The other is that it allows the task definition to be overridden for each launch request. For example a user could create a new task definition for every launch of the same app. I'm not sure we want to go there.  My preference is to stay with this opinionated approach for now. If the community needs more flexibility, we can revisit it then.

Reply:
My primary concern with this PR.  Is that now we are now saying that task launches can be directly associated with an application registration, which blurs the difference between app registrations and task definitions..   I know this is being used by one user, but is this something we want to support for all users?

Reply:
The way I think of it is that the task definition is in most cases is only needed to associate a task definition with a task app. For example, the file ingest sample doesn't need anything else. In fact the convention I use is to use the app name as the task name (In the instructions, I specifically don't do this to make it clear that creating the task definition is not the same as registering the app - but it's natural to think of them as the same IMO). So for those cases, it's just an extra bit of boilerplate to remember, and will save me a step going forward.  I wanted to make this work be default, but agreed to make it opt-in.  My guess is that many will find this useful.  The risk is if someone needs a real task definition, e.g., for CTR, and forget to create the definition, they may inadvertently kick off a misconfigured task. Not much of a down side if you ask me.

Reply:
👍